src/AddTodoForm.js

```diff
@@ -1,22 +1,30 @@
-import React from 'react';
+import React, { useState } from "react";
 
-const TITLE_ID = 'todoTitle';
-const FORM_ID = 'addTodoForm';
+const INITIAL_TITLE_VALUE = "";
+const TITLE_ID = "todoTitle";
 
 const AddTodoForm = ({ onAddToDo }) => {
-
+    const [todoTitle, setTodoTitle] = useState(INITIAL_TITLE_VALUE);
+    
+    const handleTitleChange = (event) => {
+        const newTodoTitle = event.target.value;
+        setTodoTitle(newTodoTitle);
+    };
+    
     const handleAddTodo = (event) => {
         event.preventDefault();
-        const todoTitle = event.target.title.value;
-        onAddToDo(todoTitle);
-        document.querySelector(`#${FORM_ID}`).reset();
+        onAddToDo({
+            title: todoTitle,
+            id: Date.now()
+        });
+        setTodoTitle(INITIAL_TITLE_VALUE);
     };
-
+    
     return (
-        <form id={FORM_ID} onSubmit={handleAddTodo}>
+        <form onSubmit={handleAddTodo}>
             <label htmlFor={TITLE_ID}>Title:&nbsp;
-                <input id={TITLE_ID} name='title'/>
-                <button type='submit'>Add</button>
+                <input id={TITLE_ID} name="title" value={todoTitle} onChange={handleTitleChange}/>
+                <button type="submit">Add</button>
```

Comment:
Requires changes:
This form submit button should not be a part of the `label`. The label can either wrap the `input` or be adjacent to it with an `id` attribute that references which input it's describing. However, the submit button is not a part of the label/input relationship and instead should reside outside of the label:
```jsx
return (
    <form onSubmit={handleAddTodo}>
        <label htmlFor={TITLE_ID}>Title:&nbsp;
             <input id={TITLE_ID} name="title" value={todoTitle} onChange={handleTitleChange}/>
         </label>
         <button type="submit">Add</button>
    </form>
);
```

Reply:
Thanks! I didn't realize that. I just made the changes.